### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by neil-tan**
- Added Dropout
- Updated the training scripts to use the latest utensor-cli
- Added reference code-gen files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +58,8 @@ In preparation for code generation, you must freeze the TensorFlow model. Freezi
 You can get the output node names from the IPython notebook. This will create a `models` and `constants` directory. `models` contains the embedded code interface for making inferences in your applications, and `constants` contains the *learned* weights associated with each stage in the neural net.
 
 ```
-utensor-cli tensorflow-models/mnist_model/deep_mlp.pb
+# from project root, run:
```
Comment (mbartling) [inline README.md]: @neil-tan Do we have to have the --output-nodes still? I thought this was removed?

File: tensorflow-models/deep_mlp.py [hunk 2]
```diff
@@ -103,12 +102,12 @@ def main(_):
...
-                                       [],
-                                       out_nodes,
-                                       ["quantize_weights", "quantize_nodes"])
-      graph_path = tf.train.write_graph(quant_graph_def,
+      # # quantize the graph
+      # quant_graph_def = TransformGraph(sub_graph_def,
+      #                                  [],
+      #                                  out_nodes,
+      #                                  ["quantize_weights", "quantize_nodes"])
+      graph_path = tf.train.write_graph(sub_graph_def,
```
Comment (mbartling) [inline tensorflow-models/deep_mlp.py]: Why did you remove the graph quantization?

Comment (neil-tan) [reply tensorflow-models/deep_mlp.py]: It is moved into cli because it interfer with dropout parsing. In the long term, this give us more control with graph optimization.
